### ISSUE TITLE
Access to land boundaries gives an error

### ISSUE THREAD

**Issue by shinbunya** [noaa-ocs-modeling/adcircpy#162]
An access to land boundaries of an AdcircMesh instance appears to have an issue.

```
from adcircpy import AdcircMesh
f14file = 'path/to/fort.14'
mesh = AdcircMesh.open(f14file, crs='EPSG:4326)
mesh.land_boundaries
```
gives
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[20], line 1
----> 1 mesh.land_boundaries

File d:\Python\Python38\lib\site-packages\adcircpy\mesh\fort14.py:285, in Fort14.land_boundaries(self)
    283 @property
    284 def land_boundaries(self):
--> 285     return self.boundaries.land

File d:\Python\Python38\lib\site-packages\adcircpy\mesh\fort14.py:150, in Fort14Boundaries.land(self)
    147 @property
    148 def land(self):
    149     if not hasattr(self, '_land'):
--> 150         self._land = LandBoundaries(self._mesh, self._aggregate_boundaries('0'))
    151     return self._land

File d:\Python\Python38\lib\site-packages\adcircpy\mesh\fort14.py:189, in Fort14Boundaries._aggregate_boundaries(self, endswith)
    187         continue
    188     if ibtype.endswith(endswith):
--> 189         for bdata in list(_boundaries.values()):
    190             boundaries.update({len(boundaries) + 1: {'ibtype': ibtype, **bdata,}})
    191 return boundaries

AttributeError: 'list' object has no attribute 'values'
```

while `mesh.ocean_boundaries` seems to be working.

**Comment by SorooshMani-NOAA** [comment]
I'll take a look

**Comment by SorooshMani-NOAA** [comment]
The boundaries data read from a `fort.14` file used to be like the following when originally boundary handler object was written:
https://github.com/noaa-ocs-modeling/adcircpy/blob/94213e2141f91c012f8c8d6e5b664a80d756cb3b/adcircpy/mesh/parsers/grd.py#L71

Now it is like this:
https://github.com/noaa-ocs-modeling/adcircpy/blob/244ae8a420acfb4f22ceedc05694ff36f082f665/adcircpy/mesh/parsers/grd.py#L144

Which is causing problem for the logic originally written for handling the boundary information:
https://github.com/noaa-ocs-modeling/adcircpy/blob/94213e2141f91c012f8c8d6e5b664a80d756cb3b/adcircpy/mesh/fort14.py#L200-L212
or in it's latest form (just style changes):
https://github.com/noaa-ocs-modeling/adcircpy/blob/244ae8a420acfb4f22ceedc05694ff36f082f665/adcircpy/mesh/fort14.py#L183-L191

Now the question is has this change in how boundary data is stored reflected in other parts of the code or not. If it is, then the boundary handler needs to be updated, if it is **not** then the `fort.14` parser code needs to be fixed to store boundaries like before

**Comment by SorooshMani-NOAA** [comment]
Since the write code for the boundary was also updated in the same commit, I'll go ahead and update the boundary handler code assuming the rest of the relevant code has been correctly updated to handle new boundary storage (dict of list of dict of list instead of dict of dict of dict of list)

**Comment by SorooshMani-NOAA** [comment]
The issue seems to have been introduced in https://github.com/noaa-ocs-modeling/adcircpy/pull/121. I'm still working on a fix.

**Comment by SorooshMani-NOAA** [comment]
Another issue I noticed in #121, is the switch between `id` vs `index` of nodes. Just to clarify for future references:
- _ID_ of a node is an integer value associated with a node. It can start from any numbers and doesn't need to include all the numbers in the mesh node ID range. 
- _Index_ is the position of a specific node in the array of all nodes of the mesh. It should start from 0 and must include all numbers in the range of mesh node indices.
 
For example if we have a mesh with 4 nodes, the IDs can be 1, 12, 17, 200, while the index for the same nodes should be 0, 1, 2, 3.

Usually in normal circumstances: `node_id = node_index + 1`. Unless the mesh has been modified. After #121 this meaning of ID vs index seems to have been lost/reversed(?!).

**UPDATE**
What led me to say this was `BaseBoudnaries` class and the issue in `indexes` method of that class. I then noticed that in that class `ids` refer to _boundary ID_ and `indexes` to _node indexes_. There's also `node_id` which refer to _node IDs_. This is the implementation of `indexes` prior to #121:
https://github.com/noaa-ocs-modeling/adcircpy/blob/35286dc7136ef0e38aaeb5ae9bd186143b1994fe/adcircpy/mesh/fort14.py#L29-L37

I'll update this comment as I gather more information and if I notice any mistakes in what.

**Comment by SorooshMani-NOAA** [comment]
Note that this issue might be the root cause of #154. Fixing it might also break the fix recently added (in the same branch) for fixing plotting issue.

**Comment by SorooshMani-NOAA** [comment]
@shinbunya I have committed a new set of changes to `bugfix/meshplot` branch. These should take care of the boundary information extraction as well. Please let me know if you still notice any issue. Note still the `mesh_plot` should work fine after these changes. If this is not the case for you please let me know as well.

Thank you for testing!

**Comment by shinbunya** [comment]
Thanks @SorooshMani-NOAA !
